### PR TITLE
Bug 1310212 - Re-enable UITests on XCode 8 (Part 1)

### PR DESCRIPTION
**Pull request by npark-mozilla**

### CODE REVIEW

**Comment by npark-mozilla** [comment]
Changes made:
- Setup has dismissFirstRunUI() from BrowserUtils.
- Setup opens a new tab when URL needs to be entered.
- Teardown has resetToAboutHome() that kills all tabs and puts the to tab view
- Teardown clears all browsing data via clearPrivateData() when needed
- All font size tests are disabled since in iOS10, the hack that we were using no longer works.
- testListFiltering() in LoginManagerTests is disabled, since on simulator the screen does not update properly when filtering is applied on password fields
- tests that use deactivateAppForDuration() method in LoginManagerTests is disabled, since  https://github.com/kif-framework/KIF/issues/707
- When disabling tests, commenting out was used instead of disabling them in XCode- the latter will still run the test, but just ignore the results.

File: UITests/FindInPageTests.swift [hunk 1]
```diff
@@ -90,22 +99,22 @@ class FindInPageTests: KIFTestCase {
...
+    
     func testOpenFindInPageFromMenu() {
         let testURL = "\(webRoot)/findPage.html"
-
+        
         tester().tapViewWithAccessibilityIdentifier("url")
         tester().clearTextFromAndThenEnterTextIntoCurrentFirstResponder("\(testURL)\n")
         tester().waitForWebViewElementWithAccessibilityLabel("nullam")
-
+
```
Comment (farhanpatel) [inline UITests/FindInPageTests.swift]: All the extra space like this needs to be removed please!

File: UITests/LoginManagerTests.swift [hunk 2]
```diff
@@ -65,49 +70,72 @@ class LoginManagerTests: KIFTestCase {
...
+        
         // Filter by something that doesn't match anything
         tester().waitForViewWithAccessibilityLabel("a0@email.com, http://a0.com")
         tester().tapViewWithAccessibilityLabel("Enter Search Mode")
         tester().enterTextIntoCurrentFirstResponder("thisdoesntmatch")
-
-        // TODO: Check for empty view
-
+        tester().waitForViewWithAccessibilityIdentifier("Login List")
+
```
Comment (farhanpatel) [inline UITests/LoginManagerTests.swift]: Make sure to have a max of 1 new line between code.

File: UITests/LoginManagerTests.swift [hunk 3]
```diff
@@ -65,49 +70,72 @@ class LoginManagerTests: KIFTestCase {
...
+        tester().waitForAnimationsToFinish()
+        tester().enterTextIntoCurrentFirstResponder("d9")
         list = tester().waitForViewWithAccessibilityIdentifier("Login List") as! UITableView
+        tester().waitForViewWithAccessibilityLabel("d9@email.com")
         XCTAssertEqual(list.numberOfRowsInSection(0), 1)
-
+        
         tester().tapViewWithAccessibilityLabel("Clear Search")
-
+
```
Comment (farhanpatel) [inline UITests/LoginManagerTests.swift]: Please remove the extra space.

File: UITests/LoginManagerTests.swift [hunk 4]
```diff
@@ -65,49 +70,72 @@ class LoginManagerTests: KIFTestCase {
         profile.logins.removeAll().value
     }
 
+    // This test will fail on simulator, but passes on the device because of the KIFTest issue.
+    // when the password field is filtered on, it will not update the list
+
```
Comment (farhanpatel) [inline UITests/LoginManagerTests.swift]: Remove this new line. Comments can be right above functions.

**Pull request merged**